src/main/java/gov/cms/madie/models/validators/TestCasePopulationValueValidator.java

```diff
@@ -0,0 +1,69 @@
+package gov.cms.madie.models.validators;
+
+import java.util.List;
+
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import gov.cms.madie.models.measure.MeasureScoring;
+import gov.cms.madie.models.measure.PopulationType;
+import gov.cms.madie.models.measure.TestCaseGroupPopulation;
+import gov.cms.madie.models.measure.TestCasePopulationValue;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class TestCasePopulationValueValidator
+    implements ConstraintValidator<ValidTestCasePopulationValue, TestCaseGroupPopulation> {
+
+  @Override
+  public boolean isValid(
+      TestCaseGroupPopulation testCaseGroupPopulation, ConstraintValidatorContext context) {
+    if (testCaseGroupPopulation == null
+        || CollectionUtils.isEmpty(testCaseGroupPopulation.getPopulationValues())) {
+      return true;
+    }
+    MeasureScoring measureScoring =
+        MeasureScoring.valueOfText(testCaseGroupPopulation.getScoring());
+
+    boolean isValid = true;
+    switch (measureScoring) {
+      case CONTINUOUS_VARIABLE:
+        List<TestCasePopulationValue> measurePopulationObservations =
+            testCaseGroupPopulation.getPopulationValues().stream()
+                .filter(
+                    value ->
+                        PopulationType.MEASURE_POPULATION_OBSERVATION
+                                .name()
+                                .equalsIgnoreCase(value.getName().name())
+                            && value.getExpected() != null
+                            && StringUtils.hasText(value.getExpected().toString())
+                            && (!StringUtils.hasText(value.getCriteriaReference())
+                                || !StringUtils.hasText(value.getId())))
+                .toList();
+        isValid = CollectionUtils.isEmpty(measurePopulationObservations);
+        break;
+      case RATIO:
+        List<TestCasePopulationValue> denomAndNumerObservations =
+            testCaseGroupPopulation.getPopulationValues().stream()
+                .filter(
+                    value ->
+                        (PopulationType.DENOMINATOR_OBSERVATION
+                                    .name()
+                                    .equalsIgnoreCase(value.getName().name())
+                                || PopulationType.NUMERATOR_OBSERVATION
+                                    .name()
+                                    .equalsIgnoreCase(value.getName().name()))
+                            && value.getExpected() != null
+                            && StringUtils.hasText(value.getExpected().toString())
+                            && (!StringUtils.hasText(value.getCriteriaReference())
+                                || !StringUtils.hasText(value.getId())))
+                .toList();
+        isValid = CollectionUtils.isEmpty(denomAndNumerObservations);
```

Comment:
Not a big deal, but looks like we could just return the output of `CollectionUtils.isEmpty` and doesn't have to worry about updating isValid and breaking the flow.

Reply:
default could just return true;

Reply:
@RohitKandimalla Thanks so much Rohit! I tend to use one variable and do one return instead of multiple returns in a block. I guess you are right not to use/update isValid as it's less code.